stellar-auth/pkg/auth/auth.go

```diff
@@ -209,6 +210,29 @@ func (am *defaultAuthManager) ResetPassword(ctx context.Context, resetToken, new
 	return nil
 }
 
+func (am *defaultAuthManager) UpdatePassword(ctx context.Context, tokenString, currentPassword, newPassword string) error {
+	isValid, err := am.ValidateToken(ctx, tokenString)
+	if err != nil {
+		return fmt.Errorf("validating token: %w", err)
+	}
+
+	if !isValid {
+		return ErrInvalidToken
+	}
+
+	user, err := am.jwtManager.GetUserFromToken(ctx, tokenString)
+	if err != nil {
+		return fmt.Errorf("error getting user from token: %w", err)
+	}
+
+	err = am.authenticator.UpdatePassword(ctx, user, currentPassword, newPassword)
+	if err != nil {
+		return fmt.Errorf("error updating password: %w", err)
+	}
```

Comment:
💡 nit: error messages don't need to start with `error`. Otherwise, the stack trace would look like `error doing A: error doing B, error doing C: ...`m which is kind of redundant.

The libraries usually don't start the error messages with `error` either.